redis/src/test/scala/zio/redis/ApiSpec.scala

```diff
@@ -268,6 +271,354 @@ object ApiSpec extends BaseSpec {
+          for {
+            key     <- uuid
+            value   <- uuid
+            _       <- set(key, value, None, None, None)
+            push    <- rPush(key)("hello").either
+          } yield assert(push)(isLeft)
+        },
+        testM("rPushX element") {
+          for {
+            key     <- uuid
+            _       <- rPush(key)("world")
+            px      <- rPushX(key)("hello")
+          } yield assert(px)(equalTo(2l))
+        },
+        testM("rPushX nothing when key doesn't exist") {
+          for {
+            key     <- uuid
+            px      <- rPushX(key)("world")
+          } yield assert(px)(equalTo(0l))
+        },
+        testM("rPushX error when not list") {
+          for {
+            key     <- uuid
+            value   <- uuid
+            _       <- set(key, value, None, None, None)
+            push    <- rPushX(key)("hello").either
+          } yield assert(push)(isLeft)
+        },
+        testM("rPopLPush element") {
+          for {
+            key     <- uuid
+            dest    <- uuid
+            _       <- rPush(key)("one", "two", "three")
+            _       <- rPush(dest)("four")
+            _       <- rPopLPush(key, dest)
+            r       <- lRange(key, Range(0, -1))
+            l       <- lRange(dest, Range(0, -1))
+          } yield assert(r)(equalTo(Chunk("one", "two"))) && assert(l)(equalTo(Chunk("three", "four")))
+        },
+        testM("rPopLPush nothing when source does not exist") {
+          for {
+            key     <- uuid
+            dest    <- uuid
+            _       <- rPush(dest)("four")
+            _       <- rPopLPush(key, dest)
+            l       <- lRange(dest, Range(0, -1))
+          } yield assert(l)(equalTo(Chunk("four")))
+        },
+        testM("rPopLPush error when not list") {
+          for {
+            key     <- uuid
+            dest    <- uuid
+            value   <- uuid
+            _       <- set(key, value, None, None, None)
+            rpp     <- rPopLPush(key, dest).either
+          } yield assert(rpp)(isLeft)
+        },
+        testM("brPopLPush element") {
+          for {
+            key     <- uuid
+            dest    <- uuid
+            _       <- rPush(key)("one", "two", "three")
+            _       <- rPush(dest)("four")
+            _       <- brPopLPush(key, dest, Duration(1, TimeUnit.SECONDS))
+            r       <- lRange(key, Range(0, -1))
+            l       <- lRange(dest, Range(0, -1))
+          } yield assert(r)(equalTo(Chunk("one", "two"))) && assert(l)(equalTo(Chunk("three", "four")))
+        },
+        testM("brPopLPush block for 1 second when source does not exist") {
+          for {
+            key     <- uuid
+            dest    <- uuid
+            _       <- rPush(dest)("four")
+            st      <-  currentTime(TimeUnit.SECONDS)
+            s       <- brPopLPush(key, dest, Duration(1, TimeUnit.SECONDS)).either
+            _       <- TestClock.adjust(1.seconds)
```

Comment:
I think you can drop the adjustment of a test clock. If I'm not mistaken, `currentTime` should use the live clock we are providing, which is the correct one to use in this case.